src/codegen/llvm/codegen_llvm_visitor.hpp

```diff
@@ -96,6 +99,35 @@ class CodegenLLVMVisitor: public visitor::ConstAstVisitor {
         , builder(*context)
         , fpm(module.get()) {}
 
+    /**
+     * Checks if array index specified by the given IndexedName is within bounds
+     * \param node IndexedName representing array
+     * \return     \c true if the index is within bounds
+     */
+    bool check_array_bounds(const ast::IndexedName& node, unsigned index);
+
+    /**
+     * Generates LLVM code for the given IndexedName
+     * \param node IndexedName NMODL AST node
+     * \return LLVM code generated for this AST node
+     */
+    llvm::Value* codegen_indexed_name(const ast::IndexedName& node);
+
+    /**
+     * Returns GEP instruction to 1D array
+     * \param name 1D array name
+     * \param index element index
+     * \return GEP instruction value
+     */
+    llvm::Value* create_GEP(const std::string& name, unsigned index);
```

Comment:
nitpick : stylistically, in nmodl we would write as `create_gep` (but it's true that in LLVM method has `GEP`)